### PR TITLE
Add GitHub Action to run all tests

### PR DESCRIPTION
**Pull request by alexgleith**
This adds a GitHub action that installs all the apps and libs and runs all the tests.

I have disabled some tests, which @sachitrajbhandari may be able to help with?

### CODE REVIEW

File: tests/requirements-apt.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+libyaml-dev
```
Comment (Kirill888) [inline tests/requirements-apt.txt:1]: do we really need this indirection via file?
at least rename it to `apt.txt` it has nothing to do with python requirements.txt

File: tests/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+--pre
```
Comment (Kirill888) [inline tests/requirements.txt]: no, never use `--pre` for global installs only for specific libraries

**Review by tebadi — APPROVED** [review]

File: tests/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ENV PATH="/env/bin:${PATH}"
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Copy the live code in
+ADD ./ /code
+WORKDIR /code
+
+# Copy in the test runner script
+COPY ./tests/run_tests.sh /usr/local/bin/run_tests.sh
```
Comment (Kirill888) [inline tests/Dockerfile:39]: This Docker doesn't make sense to me. Code is copied twice but installed once.

I recommend following approach:
1. build wheels outside of Docker, see other action that does that
2. Create environment using those wheels
3. Copy to runner
4. Run tests in runner

That way we test packaging as well as code, so any mistakes/missing files in packaging are more likely to be caught.

Does it even need to be 2 stage docker?

File: tests/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+--extra-index-url https://packages.dea.ga.gov.au/
+setuptools
+datacube[performance,s3]==1.8.3
+aiobotocore[boto3,awscli]==1.1.1
+aiohttp==3.6.2
+pandas==1.0.5
+digitalearthau
```
Comment (Kirill888) [inline tests/requirements.txt]: what is using `digitalearthau`?

File: tests/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+--extra-index-url https://packages.dea.ga.gov.au/
+setuptools
+datacube[performance,s3]==1.8.3
+aiobotocore[boto3,awscli]==1.1.1
+aiohttp==3.6.2
+pandas==1.0.5
```
Comment (Kirill888) [inline tests/requirements.txt]: No need to pin pandas to older version anymore, let's test with the same setup we have in sandbox currently,

```
pandas==1.1.2
xarray==0.16.1
```

File: tests/requirements-apt.txt (hunk 1 above)
Comment (alexgleith) [reply tests/requirements-apt.txt:1]: This is pretty standard in our other repositories.

File: tests/requirements.txt (hunk 4 above)
Comment (alexgleith) [reply tests/requirements.txt]: Not sure, actually

**Review by Kirill888 — APPROVED** [review]

**Pull request merged**